lib/membrane_rtmp_plugin/rtmp_server/config.ex

```diff
@@ -0,0 +1,363 @@
+      defaults: defaults,
+      app_config: app_config,
+      runtime: runtime_opts,
+      final: final
+    }
+  end
+
+  # Private functions
+
+  @spec get_default_ssl_options() :: ssl_options()
+  defp get_default_ssl_options() do
+    [
+      verify: :verify_none,
+      fail_if_no_peer_cert: false,
+      # Use only TLS 1.2 for better compatibility
+      versions: [:"tlsv1.2"],
+      secure_renegotiate: true,
+      reuse_sessions: true,
+      # More verbose logging to debug handshake issues
+      log_level: :info
+    ]
+  end
+
+  @spec get_app_config_ssl_options() :: ssl_options()
+  defp get_app_config_ssl_options() do
+    Application.get_env(:membrane_rtmp_plugin, :ssl, [])
+  end
+
+  @spec process_certificate_paths(ssl_options()) :: ssl_options()
+  defp process_certificate_paths(opts) do
+    opts
+    |> expand_certificate_paths()
+    |> resolve_relative_paths()
+  end
+
+  @spec expand_certificate_paths(ssl_options()) :: ssl_options()
+  defp expand_certificate_paths(opts) do
+    opts
+    |> maybe_expand_path(:certfile)
+    |> maybe_expand_path(:keyfile)
+    |> maybe_expand_path(:cacertfile)
+    |> maybe_expand_path(:certchain)
+  end
+
+  @spec maybe_expand_path(ssl_options(), atom()) :: ssl_options()
+  defp maybe_expand_path(opts, key) do
+    case opts[key] do
+      nil -> opts
+      path when is_binary(path) -> Keyword.put(opts, key, Path.expand(path))
+      _other -> opts
+    end
+  end
+
+  @spec resolve_relative_paths(ssl_options()) :: ssl_options()
+  defp resolve_relative_paths(opts) do
+    # If certificate files are specified with relative paths,
+    # try to resolve them relative to the app's priv directory
+    priv_dir = Application.app_dir(:membrane_rtmp_plugin, "priv")
+
+    opts
+    |> maybe_resolve_relative_to_priv(:certfile, priv_dir)
+    |> maybe_resolve_relative_to_priv(:keyfile, priv_dir)
+    |> maybe_resolve_relative_to_priv(:cacertfile, priv_dir)
+    |> maybe_resolve_relative_to_priv(:certchain, priv_dir)
+  end
+
+  @spec maybe_resolve_relative_to_priv(ssl_options(), atom(), String.t()) :: ssl_options()
+  defp maybe_resolve_relative_to_priv(opts, key, priv_dir) do
+    case opts[key] do
+      nil ->
+        opts
+
+      path when is_binary(path) ->
+        resolve_path_relative_to_priv(opts, key, path, priv_dir)
+
+      _other ->
+        opts
+    end
+  end
+
+  @spec resolve_path_relative_to_priv(ssl_options(), atom(), String.t(), String.t()) ::
+          ssl_options()
+  defp resolve_path_relative_to_priv(opts, key, path, priv_dir) do
+    if Path.absname(path) == path do
+      # Already absolute
+      opts
+    else
+      priv_path = Path.join(priv_dir, path)
+
+      if File.exists?(priv_path) do
+        Keyword.put(opts, key, priv_path)
+      else
+        # Keep original path
+        opts
+      end
+    end
+  end
+
+  @spec validate_certificate_files(ssl_options(), boolean()) :: ssl_options()
+  defp validate_certificate_files(opts, validate_files) do
```

Comment:
Couldn't we just pattern match on` validate_files` being `true` here and directly call `raise ArgumentError` (instead of `validate_singe_cert_file` etc.) in case some file is missing?